pkg/controller/common/rollout/workloads/statefulset_controller.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2021 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package workloads
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	apps "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/oam-dev/kubevela/apis/core.oam.dev/v1beta1"
+)
+
+type statefulSetController struct {
+	workloadController
+	targetNamespacedName types.NamespacedName
+}
+
+// add the parent controller to the owner of the StatefulSet, and initialize the size
+// before kicking start the update and start from every pod in the old version
+func (c *statefulSetController) claimStatefulSet(ctx context.Context, statefulSet *apps.StatefulSet, initSize *int32) (bool, error) {
+	if controller := metav1.GetControllerOf(statefulSet); controller != nil &&
+		controller.Kind == v1beta1.AppRolloutKind && controller.APIVersion == v1beta1.SchemeGroupVersion.String() {
+		// it's already there
+		return true, nil
+	}
+
+	statefulSetPatch := client.MergeFrom(statefulSet.DeepCopyObject())
+
+	// add the parent controller to the owner of the StatefulSet
+	ref := metav1.NewControllerRef(c.parentController, v1beta1.AppRolloutKindVersionKind)
```

Comment:
please replace `v1beta1.AppRolloutKindVersionKind` with `c.parentController.GetObjectKind().GroupVersionKind()` we will support more kind of object  as parent

Reply:
It seems like `c.parentController.GetObjectKind().GroupVersionKind()` would return an empty `GroupVersionKind` struct:

```
GroupVersionKind{
    Group:   "",
    Version: "",
    Kind:    "",
}
```

Is there another way to do that?

Reply:
I reproduce this issue.I think this a test-side issue, in integration-test we generate a fake appRollout that not exist in k8s . so these fields are empty. You needn't modify it , i will fix it in my pr.